Automattic/pocket-casts-android
# Refactor bookmark add deep linking

## Description

> [!note]
> This will be a series of PRs that move deep linking to a separate module that can be tested. The goal is to address #2405 but I don't want to do it blindly without having tests in order to not introduce regressions to deep linking.

This PR migrates adding bookmark linking to the new module.

## Testing Instructions

You can test it by assigning `Add bookmark` action to your headphones and triggering it while app is in the foreground. If you can't test it this way do the following:

1. Play an episode.
2. Execute `adb shell am start -n au.com.shiftyjelly.pocketcasts.debug/au.com.shiftyjelly.pocketcasts.ui.MainActivity -a INTENT_OPEN_APP_ADD_BOOKMARK`.
3. The add bookmark page should show.

## Checklist
